lib/main/atom/commands/commands.ts

```diff
@@ -99,6 +101,26 @@ export function registerCommands() {
 
         parent.build({ filePath: filePath }).then((resp) => {
             buildView.setBuildOutput(resp.buildOutput);
+            let proj = getOrCreateProject(filePath);
```

Comment:
Sorry I just re-read the code. *Don't call `getOrCreateProject` from the UI process*. This will double our memory consumption and make the UI much slower. Instead do this stuff in the `build` function's response (feel free to add a property there) and just use the response in the UI thread.

Some docs on architecture : https://github.com/TypeStrong/atom-typescript/blob/master/CONTRIBUTING.md#architecture 

Before I forget : Thanks for the PR! :rose: